client/src/interface.rs

```diff
@@ -569,4 +632,231 @@ impl Stronghold {
+        let vault_path = vault_path.to_vec();
+        // check if vault exists
+        let b = match self
+            .ask_remote(peer_id, SHRequest::CheckVault(vault_path.clone()))
+            .await
+        {
+            CommunicationResults::RequestMsgResult(Ok(SHResults::ReturnExistsVault(b))) => b,
+            other => return Stronghold::handle_response_failure(other, "write data"),
+        };
+        if b {
+            // check if record exists
+            let b = match self
+                .ask_remote(
+                    peer_id,
+                    SHRequest::CheckRecord {
+                        location: location.clone(),
+                    },
+                )
+                .await
+            {
+                CommunicationResults::RequestMsgResult(Ok(SHResults::ReturnExistsRecord(b))) => b,
+                other => return Stronghold::handle_response_failure(other, "check record"),
+            };
+            if !b {
+                // initialize a new record
+                match self
+                    .ask_remote(
+                        peer_id,
+                        SHRequest::InitRecord {
+                            location: location.clone(),
+                        },
+                    )
+                    .await
+                {
+                    CommunicationResults::RequestMsgResult(Ok(SHResults::ReturnInitRecord(status))) => status,
+                    other => return Stronghold::handle_response_failure(other, "initialize record"),
+                };
+            }
+        } else {
+            // no vault so create new one before writing.
+            match self
+                .ask_remote(peer_id, SHRequest::CreateNewVault(location.clone()))
+                .await
+            {
+                CommunicationResults::RequestMsgResult(Ok(SHResults::ReturnCreateVault(_))) => {}
+                other => return Stronghold::handle_response_failure(other, "create vault"),
+            };
+        }
+        // write data
+        match self
+            .ask_remote(
+                peer_id,
+                SHRequest::WriteToVault {
+                    location: location.clone(),
+                    payload: payload.clone(),
+                    hint,
+                },
+            )
+            .await
+        {
+            CommunicationResults::RequestMsgResult(Ok(SHResults::ReturnWriteVault(status))) => status,
+            other => Stronghold::handle_response_failure(other, "write data"),
+        }
+    }
+
+    #[cfg(feature = "communication")]
+    /// Write to the store of a remote stronghold
+    pub async fn write_to_remote_store(
+        &self,
+        peer_id: PeerId,
+        addr: Multiaddr,
+        location: Location,
+        payload: Vec<u8>,
+        lifetime: Option<Duration>,
+    ) -> StatusMessage {
+        // connect remote peer
+        let peer_id = match ask::<_, _, CommunicationResults<SHResults>, _>(
```

Comment:
this `ConnectPeer` block repeats a lot, maybe it should be a separate function too ;)